### PR TITLE
Non-allocating VectorContinuousCallbacks and better typing

### PR DESCRIPTION
**Pull request by ChrisRackauckas**
Works towards https://discourse.julialang.org/t/significant-allocations-with-callbacks-tsit5/66467

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
Related to https://github.com/SciML/DifferentialEquations.jl/issues/786 on callbacks.

**Comment by ChrisRackauckas** [comment]
https://github.com/search?q=org%3ASciML+max_vector_callback_length&type=code should all get updated to the new `int` version.

File: src/callbacks.jl [hunk 1]
```diff
@@ -641,9 +641,27 @@ function bisection(f, tup, t_forward::Bool, rootfind::RootfindOpt, abstol, relto
...
+"""
+function findall_events!(next_sign::Union{Array,SubArray},affect!::F1,affect_neg!::F2,prev_sign::Union{Array,SubArray}) where {F1,F2}
+  @inbounds for i in 1:length(prev_sign)
+    next_sign[i] = ((prev_sign[i] < 0 && affect! !== nothing) || (prev_sign[i] > 0 && affect_neg! !== nothing)) && prev_sign[i]*next_sign[i]<=0
+  end
+  next_sign
+end
+
+function findall_events!(next_sign,affect!::F1,affect_neg!::F2,prev_sign) where {F1,F2}
+  @show typeof(next_sign)
```
Comment (KristofferC) [inline src/callbacks.jl:658]: Just pointing this one out.

Comment (ChrisRackauckas) [reply src/callbacks.jl:658]: you're a saint.

**Pull request merged**